typedspark/_core/datatypes.py

```diff
@@ -79,3 +81,22 @@ class DecimalType(Generic[_Precision, _Scale], TypedSparkDataType):
         class Numbers(Schema):
             number: Column[DecimalType[Literal[10], Literal[0]]]
     """
+
+
+class IntervalType:
+    """Interval types for ``DayTimeIntervalType``."""
+
+    DAY = Literal[0]
+    HOUR = Literal[1]
+    MINUTE = Literal[2]
+    SECOND = Literal[3]
+
+
+class DayTimeIntervalType(Generic[_StartField, _EndField], TypedSparkDataType):
+    """Allows for type annotations such as.
+
+    .. code-block:: python
+
+        class TimeInterval(Schema):
+            interval: Column[DayTimeIntervalType[Literal[0], Literal[10]]]
```

Comment:
Is `Literal[10]` even possible here? It seems from reading the `IntervalType` class that it's just values 0-3

Reply:
Oops, that's Copilot writing my docstrings there :p